feat: go-ipfs 0.5 release (#402)
ipfs/blog: content/post/090-go-ipfs-0-5-0.md

---
date: 2020-04-28
url: 2020-04-28-go-ipfs-0-5-0
title: IPFS 0.5.0 is here! Our largest upgrade to IPFS yet.
author: Molly Mackinlay
header_image: 090-go-ipfs-0-5-0.png
tags: go-ipfs, release
---

> ![Announcing IPFS 0.5](static/header_images/090-go-ipfs-0-5-0.png)

Reviewer: ```suggestion
![Announcing IPFS 0.5](static/header_images/090-go-ipfs-0-5-0.png)

```

*go-ipfs 0.5.0 is a major step forward on our journey toward a more efficient, secure, and resilient web to preserve and grow humanity’s knowledge.* 

We’re happy to share that IPFS 0.5.0, the largest upgrade to the IPFS protocol and Public Network yet, is now available. The result of extensive community feedback and rigorous focus, IPFS 0.5.0 offers major performance and reliability updates so that you can distribute content peer-to-peer more efficiently and affordably than ever before.

> 

Reviewer: ```suggestion
```

***Ian Preston, Peergos:*** “What we’ve seen is that an inactive IPFS instance now uses way less bandwidth to maintain the DHT and other background traffic. With IPFS 0.5, this load has gone down by a factor of four: it was about 800 kilobits a second and now it’s 200 kilobits a second. That doesn’t sound like much, but it adds up fast - so this makes a big difference.”

***Brendan O’Brien, Qri:*** “Our engineering time is one of our most valuable resources. We’re now seeing a much more mature version of IPFS, which means we're able to spend more time on our own code vs our dependencies. It took less than a day to upgrade to IPFS 0.5.0, allowing us to commit more of our engineering time to building and fixing our own things.”

> If you’re running IPFS, we highly encourage you to upgrade ASAP to take advantage of all these improvements! Check out our [release notes] and [upgrade how-to] to get started today!

Reviewer: ```suggestion
If you’re running IPFS, we highly encourage you to upgrade ASAP to take advantage of all these improvements! Check out our [release notes](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/) and [upgrade how-to](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/update-procedure) to get started today!
```

## Thank you, thank you, and thank you.

There is no way IPFS 0.5.0 could have been our biggest update yet without the help of many contributors.  A special thanks to our [early testers](https://github.com/ipfs/go-ipfs/blob/master/docs/EARLY_TESTERS.md) for this release, especially: 

- @MichaelMure (Infura)
- @carsonfarmer (Textile)
- @b5 (Qri)
- @obo20 (Pinata)
- @ianopolous (Peergos)
- @postables (RTrade)
- @meeDamian
- @pataquets

Also, a huge thank you to [everyone who contributed](changelog link) patches and improvements in this release, all XXX of you! We couldn’t have made this happen without your help and feedback.

> [Image: Contributors]

Reviewer: todo

## Want to learn more...or even, better get involved?

We’d like to know what you think and help answer any questions you may have. Please join us for a virtual meetup on Thursday, April 30th, to deep-dive on the improvements in this release and field questions and feedback. 

> 

Reviewer: todo add meetup invite image

> In the meantime, get started by [installing] or [upgrading] to go-ipfs 0.5 to see the improvements yourself! 

Reviewer: ```suggestion
In the meantime, get started by [installing go-ipfs](http://localhost:8080/ipns/dist.ipfs.io/#go-ipfs) or [upgrading](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/update-procedure) to go-ipfs 0.5 to see the improvements yourself! 
```

> In the meantime, get started by [installing go-ipfs](http://localhost:8080/ipns/dist.ipfs.io/#go-ipfs) or [upgrading to go-ipfs 0.5](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/update-procedure) to see the improvements yourself! 

Reviewer: ```suggestion
In the meantime, get started by [installing go-ipfs](https://dist.ipfs.io/#go-ipfs) or [upgrading to go-ipfs 0.5](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/update-procedure) to see the improvements yourself! 
```

---
date: 2020-04-28
url: 2020-04-28-go-ipfs-0-5-0

> title: IPFS 0.5.0 is here! Our largest upgrade to IPFS yet.

Reviewer: ```suggestion
title: IPFS 0.5.0 is here! Our largest upgrade to IPFS yet
```

author: Molly Mackinlay
header_image: 090-go-ipfs-0-5-0.png
tags: go-ipfs, release
---
![Announcing IPFS 0.5](static/header_images/090-go-ipfs-0-5-0.png)

*go-ipfs 0.5.0 is a major step forward on our journey toward a more efficient, secure, and resilient web to preserve and grow humanity’s knowledge.* 

> We’re happy to share that IPFS 0.5.0, the largest upgrade to the IPFS protocol and Public Network yet, is now available. The result of extensive community feedback and rigorous focus, IPFS 0.5.0 offers major performance and reliability updates so that you can distribute content peer-to-peer more efficiently and affordably than ever before.

Reviewer: ```suggestion
We’re happy to share that IPFS 0.5.0, the largest upgrade to the IPFS protocol and Public Network yet, is now available. The result of extensive community feedback and rigorous focus, IPFS 0.5.0 offers major performance and reliability updates so that you can distribute content peer-to-peer more efficiently and affordably.
```

[Check out our changelog](https://github.com/ipfs/go-ipfs/blob/master/CHANGELOG.md) if you’d like to dive right into the details.

## What IPFS 0.5.0 means for the web

Today’s web is fragile, inefficient, and expensive. It’s built on a client-server model that finds and serves up content based on its location on a particular host — no matter how far away or inaccessible. It is a system that can no longer stand up to the highly dynamic ways we communicate, create, work, consume, and connect.

The distributed web is changing that. IPFS rearchitects the web to work peer to peer, addressing data by **what** it is instead of **where** it’s located. In doing so, IPFS powers a resilient and scalable network. Our latest release, IPFS 0.5.0, offers even greater performance and reliability to meet the increasing demands on the web today, as well as into the future.

## Community feedback was our roadmap

The IPFS Public Network has experienced substantial growth (scaling 30x in 2019!), bringing new opportunities for IPFS users and the wider InterPlanetary ecosystem. Major deployments and applications that require fast and reliable connectivity have started building on and supporting IPFS:

> * **Opera & Brave:** [Opera](https://blog.ipfs.io/2020-03-30-ipfs-in-opera-for-android/) became the first major web browser to offer default IPFS support on Android, shortly after [Brave](https://blog.ipfs.io/2019-10-08-ipfs-browsers-update/) started directly embedding the IPFS Companion extension (complete with a built-in js-ipfs node). This means millions of people around the world now have access to the decentralized web built directly into their browsers.

Reviewer: ```suggestion

* **Opera & Brave:** [Opera](https://blog.ipfs.io/2020-03-30-ipfs-in-opera-for-android/) became the first major web browser to offer default IPFS support on Android, shortly after [Brave](https://blog.ipfs.io/2019-10-08-ipfs-browsers-update/) started directly embedding the IPFS Companion extension (complete with a built-in js-ipfs node). This means millions of people around the world now have access to the decentralized web built directly into their browsers.
```

## But wait—there’s more!

We said this was a major release, after all. 😉 In IPFS 0.5, we also made great progress on the developer tools and security front, including:

* **Broader support for security and encryption protocols:** IPFS now uses Transport Layer Security (TLS),  an open standard for transport security, by default to help maintain secure connections between clients and servers.
* **Developer experience improvements:**  IPFS 0.5.0 supports many existing tools and workflows, making the power of decentralization accessible to more developers. Updates include:
  * **Subdomains in the HTTP gateway:**  Developers can now put dapps (decentralized apps) in isolated origins to secure their applications and web content. This works with hashes, IPNS, DNSLink, ENS, and others.
  * **/ipfs/peer_id → /p2p/peer_id:** We now disambiguate ipfs files from peer addresses by using /p2p/ for the latter.
  * **.eth link support:**  Blockchain-based .eth links are supported by default, opening up IPFS further to decentralized application developers (support for more distributed TLDs coming soon!).

> Please see our [changelog](https://github.com/ipfs/go-ipfs/blob/master/CHANGELOG.md) for the comprehensive list of IPFS 0.5.0. updates, as well as our deep dive on all the great things in this release.

Reviewer: ```suggestion
Please see our [changelog](https://github.com/ipfs/go-ipfs/blob/master/CHANGELOG.md) for the comprehensive list of IPFS 0.5 updates, as well as our deep dive on all the great things in this release.
```

***Brendan O’Brien, Qri:*** “Our engineering time is one of our most valuable resources. We’re now seeing a much more mature version of IPFS, which means we're able to spend more time on our own code vs our dependencies. It took less than a day to upgrade to IPFS 0.5.0, allowing us to commit more of our engineering time to building and fixing our own things.”

If you’re running IPFS, we highly encourage you to upgrade ASAP to take advantage of all these improvements! Check out our [release notes](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/) and [upgrade how-to](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/update-procedure) to get started today!

## Thank you, thank you, and thank you.

There is no way IPFS 0.5.0 could have been our biggest update yet without the help of many contributors.  A special thanks to our [early testers](https://github.com/ipfs/go-ipfs/blob/master/docs/EARLY_TESTERS.md) for this release, especially: 

- @MichaelMure (Infura)
- @carsonfarmer (Textile)
- @b5 (Qri)
- @obo20 (Pinata)
- @ianopolous (Peergos)
- @postables (RTrade)
- @meeDamian
- @pataquets

> Also, a huge thank you to [everyone who contributed](https://github.com/ipfs/go-ipfs/blob/master/CHANGELOG.md) patches and improvements in this release, all **841** of you! We couldn’t have made this happen without your help and feedback. ❤

Reviewer: ```suggestion
Also, a huge thank you to [everyone who contributed](https://github.com/ipfs/go-ipfs/blob/master/CHANGELOG.md) patches and improvements in this release, all **98** of you! We couldn’t have made this happen without your help and feedback. ❤
```

> ![IPFS Contributors](static/090-go-ipfs-0-5-0/ipfs-contributors.png)

Reviewer: ```suggestion
[![IPFS Contributors](static/090-go-ipfs-0-5-0/ipfs-contributors.png)](https://github.com/ipfs/go-ipfs/blob/master/CHANGELOG.md#contributors)
```

## Want to learn more...or even, better get involved?

We’d like to know what you think and help answer any questions you may have. [Please join us for a virtual meetup](https://www.meetup.com/San-Francisco-IPFS/events/270212268/) on **Friday, May 1st**, to deep-dive on the improvements in this release and field questions and feedback. 

> ![IPFS 0.5 Launch Meetup](static/090-go-ipfs-0-5-0/Launch-Meetup.png)

Reviewer: ```suggestion
[![IPFS 0.5 Launch Meetup](static/090-go-ipfs-0-5-0/Launch-Meetup.png)](https://www.meetup.com/San-Francisco-IPFS/events/270212268/)
```

***Brendan O’Brien, Qri:*** “Our engineering time is one of our most valuable resources. We’re now seeing a much more mature version of IPFS, which means we're able to spend more time on our own code vs our dependencies. It took less than a day to upgrade to IPFS 0.5.0, allowing us to commit more of our engineering time to building and fixing our own things.”

If you’re running IPFS, we highly encourage you to upgrade ASAP to take advantage of all these improvements! Check out our [release notes](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/) and [upgrade how-to](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/update-procedure) to get started today!

## Thank you, thank you, and thank you.

There is no way IPFS 0.5.0 could have been our biggest update yet without the help of many contributors.  A special thanks to our [early testers](https://github.com/ipfs/go-ipfs/blob/master/docs/EARLY_TESTERS.md) for this release, especially: 

- @MichaelMure (Infura)
- @carsonfarmer (Textile)
- @b5 (Qri)
- @obo20 (Pinata)
- @ianopolous (Peergos)
- @postables (RTrade)
- @meeDamian
- @pataquets

> Also, a huge thank you to [everyone who contributed](https://github.com/ipfs/go-ipfs/blob/master/CHANGELOG.md) patches and improvements in this release, all **98** of you! We couldn’t have made this happen without your help and feedback. ❤

Reviewer: ```suggestion
Also, a huge thank you to [everyone who contributed](https://github.com/ipfs/go-ipfs/blob/master/CHANGELOG.md#contributors) patches and improvements in this release, all **98** of you! We couldn’t have made this happen without your help and feedback. ❤
```

---
date: 2020-04-28
url: 2020-04-28-go-ipfs-0-5-0
title: IPFS 0.5.0 is here! Our largest upgrade to IPFS yet.
author: Molly Mackinlay
header_image: 090-go-ipfs-0-5-0.png
tags: go-ipfs, release
---

> 

Reviewer: Insert header image here

Reviewer: ```suggestion
![Announcing IPFS 0.5](static/header_images/090-go-ipfs-0-5-0.png)
```

*go-ipfs 0.5.0 is a major step forward on our journey toward a more efficient, secure, and resilient web to preserve and grow humanity’s knowledge.* 

We’re happy to share that IPFS 0.5.0, the largest upgrade to the IPFS protocol and Public Network yet, is now available. The result of extensive community feedback and rigorous focus, IPFS 0.5.0 offers major performance and reliability updates so that you can distribute content peer-to-peer more efficiently and affordably than ever before.

> [Potential Video Spot]

Reviewer: ```suggestion
```

> [Check out our changelog if you’d like to dive right into the details.]

Reviewer: To link

## Community Feedback Was Our Roadmap

The IPFS Public Network has experienced substantial growth (scaling 30x in 2019!), bringing new opportunities for IPFS users and the wider InterPlanetary ecosystem. Major deployments and applications that require fast and reliable connectivity have started building on and supporting IPFS:
* **Opera & Brave:** [Opera](https://blog.ipfs.io/2020-03-30-ipfs-in-opera-for-android/) became the first major web browser to offer default IPFS support on Android, shortly after [Brave](https://blog.ipfs.io/2019-10-08-ipfs-browsers-update/) started directly embedding the IPFS Companion extension (complete with a built-in js-ipfs node). This means millions of people around the world now have access to the decentralized web built directly into their browsers.
* **Microsoft ION:**  [Microsoft has turned to IPFS](https://techcommunity.microsoft.com/t5/azure-active-directory-identity/toward-scalable-decentralized-identifier-systems/ba-p/560168) as the content addressed storage layer to power its take on a decentralized identity system.

> * **Netflix:**  [We began collaborating](https://blog.ipfs.io/2020-02-14-improved-bitswap-for-container-distribution/) on ways to incorporate peer-to-peer services into Netflix’s developer tooling to speed up their massive container layer distribution.

Reviewer: ```suggestion
* **Netflix:**  [We began collaborating](https://blog.ipfs.io/2020-02-14-improved-bitswap-for-container-distribution/) on ways to incorporate peer-to-peer services into Netflix’s developer tooling to speed up container image distribution in their build pipeline.
```

Here are a few of the improvements made in IPFS 0.5.0:

![Improvements to go-ipfs 0.5.0](static/090-go-ipfs-0-5-0/0.5-Infographic.png)

* **Faster content discovery:**  IPFS 0.5.0 makes finding and sharing data in the network much faster, thanks to large improvements to our DHT logic. More efficient routing and providing—already 2-3x faster—lowers bandwidth usage and background traffic. We’ve also updated our experimental QUIC transport to reduce connection latency and round-trips - with the aim to make this the default next release!
* **Speedy files transfers:** We sped up our file transfer mechanism, Bitswap, to make sharing data between many nodes much faster. [Read more](https://blog.ipfs.io/2020-02-14-improved-bitswap-for-container-distribution/) about how we worked with Netflix to benchmark and stress-test these improvements!
* **Faster file adding:** It’s now up to 2x faster to add files to the IPFS network using the Badger datastore on Linux and MacOS. Huzzah for async writes! We’ve also configured Badger to use less memory.

> * **Faster and more reliable mutable links:** The Inter-Planetary Name System (IPNS), our system for creating mutable links for content addresses, now provides faster naming lookups and has a new experimental pubsub transport to speed up record distribution. Providing an IPNS record is now 30-40x faster in our test network!

Reviewer: ```suggestion
* **Faster and more reliable mutable links:** The Inter-Planetary Name System (IPNS), our system for creating mutable links for content addresses, now provides faster naming lookups and has a new experimental pubsub transport to speed up record distribution. Providing an IPNS record is now _30-40x_ faster in 1K node network simulations!
```

> To ensure the reliability of all these large-scale improvements, we created [Testground], a new testing tool for peer-to-peer networks at scale. This allowed us to validate and benchmark each enhancement across many network configurations with 1K+ node simulations - increasing development velocity and rigor.

Reviewer: ```suggestion
To ensure the reliability of all these large-scale improvements, we created [Testground](http://testground.ipfs.team/), a new testing tool for peer-to-peer networks at scale. This allowed us to validate and benchmark each enhancement across many network configurations with 1K+ node simulations - increasing development velocity and rigor.
```

## But wait - there’s more!

We said this was a major release, after all. 😉 In IPFS 0.5, we also made great progress on the developer tools and security front, including:

* **Broader support for security and encryption protocols:** IPFS now uses Transport Layer Security (TLS),  an open standard for transport security, by default to help maintain secure connections between clients and servers.
* **Developer experience improvements:**  IPFS 0.5.0 supports many existing tools and workflows, making the power of decentralization accessible to more developers. Updates include:

>   * **Subdomains in the HTTP gateway:**  Developers can now put dapps (decentralized apps) in isolated origins to create security isolations if they’re using DNSlink. 

Reviewer: ```suggestion
  * **Subdomains in the HTTP gateway:**  Developers can now put dapps (decentralized apps) in isolated origins to create secure site isolations if they’re using DNSlink. 
```

>   * **/ipfs/peer_id → /p2p/peer_id:** We now disambiguate ipfs files from addresses by using /p2p/ for the latter

Reviewer: ```suggestion
  * **/ipfs/peer_id → /p2p/peer_id:** We now disambiguate ipfs files from peer addresses by using /p2p/ for the latter.
```

  * **.eth link support:**  Blockchain-based .eth links are supported by default, opening up IPFS further to decentralized application developers (support for more distributed TLDs coming soon!).

> Please see our [changelog] for the comprehensive list of IPFS 0.5.0. updates, as well as our deep dive on all the great things in this release.

Reviewer: Todo link

## The Impact of IPFS 0.5.0 (already!)

With our new testing and release process, IPFS 0.5.0 has already been put to use in a variety of production environments. Here’s how it’s making a difference for IPFS users:

***Matt Ober, Pinata:*** “IPFS 0.5 makes big improvements to Bitswap, which de-dupes data transfer between nodes. This means data providers like Pinata who are transferring a lot of data all around the world every day, can send each block only once and it gets to its destination. That cuts our bandwidth costs down significantly. And, it’s also faster.”

![Improved container image performance](static/090-go-ipfs-0-5-0/container-image.png)

***Michael Godsey, Infura:*** “Our collaboration with Protocol Labs and the latest go-ipfs v0.5 release are essential to our upcoming premium IPFS product release later this year. The upgraded subdomain gateway support greatly improves the security of our offering while helping us support some great new features. The improvements to the DHT and stability of peering allow us to provide an even more reliable product and continue to get IPFS into the hands of decentralized application developers.”

> ***Ian Preston, Peergos:** “What we’ve seen is that an inactive IPFS instance now uses way less bandwidth to maintain the DHT and other background traffic. With IPFS 0.5, this load has gone down by a factor of four: it was about 800 kilobits a second and now it’s 200 kilobits a second. That doesn’t sound like much, but it adds up fast - so this makes a big difference.”

Reviewer: ```suggestion
***Ian Preston, Peergos:*** “What we’ve seen is that an inactive IPFS instance now uses way less bandwidth to maintain the DHT and other background traffic. With IPFS 0.5, this load has gone down by a factor of four: it was about 800 kilobits a second and now it’s 200 kilobits a second. That doesn’t sound like much, but it adds up fast - so this makes a big difference.”
```

***Brendan O’Brien, Qri:*** “Our engineering time is one of our most valuable resources. We’re now seeing a much more mature version of IPFS, which means we're able to spend more time on our own code vs our dependencies. It took less than a day to upgrade to IPFS 0.5.0, allowing us to commit more of our engineering time to building and fixing our own things.”

If you’re running IPFS, we highly encourage you to upgrade ASAP to take advantage of all these improvements! Check out our [release notes] and [upgrade how-to] to get started today!

## Thank you, thank you, and thank you.

> There is no way IPFS 0.5.0 could have been our biggest update yet without the help of many contributors.  A special thanks to our early testers for this release including: 

Reviewer: ```suggestion
There is no way IPFS 0.5.0 could have been our biggest update yet without the help of many contributors.  A special thanks to our early testers for this release, especially: 
```

> - Infura, Textile, Qri, Pinata, etc

Reviewer: ```suggestion
- @MichaelMure from Infura
- @carsonfarmer from Textile
- @b5 from Qri
- @obo20 from Pinata
- @ianopolous from Peergos
- @postables from RTrade
- @meeDamian
- @pataquets
```

> Also, a huge thank you to [everyone who contributed](changelog link) patches and improvements in this release, all XXX of you! We couldn’t have made this happen without your help and feedback.

Reviewer: Todo link
